src/docfx/build/metadata/ContributionProvider.cs

```diff
@@ -183,7 +183,7 @@ private PathString GetOriginalFullPath(FilePath file)
             return _input.GetFullPath(originalPath is null ? file : new FilePath(originalPath));
         }
 
-        private string? GetContentGitUrl(string repo, string branch, string pathToRepo)
+        private string? GetContentGitUrl(string repo, string? branch, string pathToRepo)
```

Comment:
This would produce an invalid URL like `blob//a.md`. Consider changing line 205 to fallback to commit hash to produce a clickable URL like `blob/a09cd/a.md` in case of detached head.